java/cli/src/main/java/org/ray/cli/RayCli.java

```diff
@@ -0,0 +1,320 @@
+      manager = startRayNode(params, paths, config);
+    }
+    return manager;
+  }
+
+  private static void start(CommandStart cmdStart, ConfigReader reader) {
+    RayParameters params = new RayParameters(reader);
+
+    RunManager manager = null;
+
+    manager = startProcess(cmdStart, reader);
+
+    // monitoring all processes, throwing an exception when any process fails
+    while (true) {
+      try {
+        Thread.sleep(5000);
+      } catch (InterruptedException e) {
+      }
+
+      HashSet<RunInfo.ProcessType> excludeTypes = new HashSet<>();
+      if (!manager.checkAlive(excludeTypes)) {
+
+        //ray components fail-over
+        RayLog.core.error("Something error in Ray processes.");
+
+        if (!manager.tryRecoverDeadProcess()) {
+          RayLog.core.error("Restart all the processes in this node.");
+          manager.cleanup(true);
+          manager = startProcess(cmdStart, reader);
+        }
+      }
+    }
+  }
+
+  private static void stop(CommandStop cmdStop) {
+    String[] cmd = {"/bin/sh", "-c", ""};
+
+    cmd[2] = "killall global_scheduler local_scheduler plasma_store plasma_manager";
+    try {
+      Runtime.getRuntime().exec(cmd);
+    } catch (IOException e) {
+    }
+
+    cmd[2] = "kill $(ps aux | grep redis-server | grep -v grep | " +
+        "awk \'{ print $2 }\') 2> /dev/null";
+    try {
+      Runtime.getRuntime().exec(cmd);
+    } catch (IOException e) {
+    }
+
+    cmd[2] = "kill -9 $(ps aux | grep DefaultWorker | grep -v grep | " +
+        "awk \'{ print $2 }\') 2> /dev/null";
+    try {
+      Runtime.getRuntime().exec(cmd);
+    } catch (IOException e) {
+    }
+  }
+
+  private static String[] buildRayRuntimeArgs(CommandSubmit cmdSubmit) {
+
+    if (cmdSubmit.redis_address == null) {
+      throw new RuntimeException(
+          "--redis-address must be specified to submit a job");      
+    }
+
+    List<String> argList = new ArrayList<String>();
+    String section = "ray.java.start.";
+    String overwrite = "--overwrite="
+      + section + "redis_address=" + cmdSubmit.redis_address + ";"
+      + section + "run_mode=" + "CLUSTER";
+
+    argList.add(overwrite);
+
+    if (cmdSubmit.config != null) {
+      String config = "--config=" + cmdSubmit.config;
+      argList.add(config);
+    }
+
+    String args[] = new String[argList.size()];
+    argList.toArray(args);
+
+    return args;
+  }
+ 
+  private static void submit(CommandSubmit cmdSubmit, String configPath) throws Exception {
+    ConfigReader config = new ConfigReader(configPath, "ray.java.start.deploy=true");
+    PathConfig paths = new PathConfig(config);
+    RayParameters params = new RayParameters(config);
+
+    params.redis_address = cmdSubmit.redis_address;
+    params.run_mode = RunMode.CLUSTER;
+
+
+    KeyValueStoreLink kvStore = new RedisClient();
+    kvStore.setAddr(cmdSubmit.redis_address);
+    StateStoreProxy stateStoreProxy = new StateStoreProxyImpl(kvStore);
+    //stateStoreProxy.setStore(kvStore);
```

Comment:
Could you also remove the commented-out code in this function? or add some comments explaining why it should be commented out.

Reply:
removed.